cpu/stm32f4/include/periph_cpu.h

```diff
@@ -84,6 +84,122 @@ typedef struct {
     uint8_t chan;           /**< CPU ADC channel connected to the pin */
 } adc_conf_t;
 
+/**
+ * @brief   Ethernet Peripheral configuration
+ */
+typedef struct {
+    enum {
+        MII = 18,                       /**< Configuration for MII */
+        RMII = 9,                       /**< Configuration for RMII */
+        SMI = 2,                        /**< Configuration for SMI */
+    } mode;                             /**< Select configuration mode */
+    enum {
+        ETH_SPEED_10T_HD = 0x0000,
+        ETH_SPEED_10T_FD = 0x0100,
+        ETH_SPEED_100TX_HD = 0x2000,
+        ETH_SPEED_100TX_FD = 0x2100,
+    } speed;                            /**< Speed selection */
+    uint8_t dma_stream;                 /**< DMA stream used for TX */
+    uint8_t dma_chan;                   /**< DMA channel used for TX */
+    char mac[6];                        /**< Et hernet MAC address */
+    char phy_addr;                      /**< PHY address */
+    gpio_t pins[];                      /**< Pins to use. MII requires 18 pins,
+                                             RMII 9 and SMI 9. Not all speeds are
+                                             supported by all modes. */
+} eth_conf_t;
+
+/**
```

Comment:
all DMA stuff shouldn't be there. There is a `periph_dma` already available (I have to re-check this). You can also have a look at #9171 (and maybe also base this PR on it ?)